src/GrasshopperTeklaDrawingLink/Components/Text/CreateTextComponent.cs

```diff
@@ -0,0 +1,169 @@
+                    pManager.AddPointParameter(ParamInfos.MarkLeaderLineEndPoint.Name, 
+                                                ParamInfos.MarkLeaderLineEndPoint.NickName, 
+                                                ParamInfos.MarkLeaderLineEndPoint.Description, 
+                                                GH_ParamAccess.list),
+                     pManager.AddParameter(new TextAttributesParam(ParamInfos.TextAttributes, GH_ParamAccess.list))
+            });
+        }
+
+        protected override void RegisterOutputParams(GH_OutputParamManager pManager)
+        {
+            pManager.AddParameter(new TeklaDatabaseObjectParam(ParamInfos.Text, GH_ParamAccess.list));
+        }
+
+
+        protected override IEnumerable<TSD.DatabaseObject> InsertObjects(IGH_DataAccess dA)
+        {
+            if (!(dA.GetGooValue<TSD.DatabaseObject>(ParamInfos.View) is TSD.View view))
+                return null;
+
+            var insertionPoints = new List<Point3d>();
+            if (!dA.GetDataList(ParamInfos.MarkInsertionPoint.Name, insertionPoints))
+                return null;
+            var leaderLineEndPoints = new List<Point3d>();
+            dA.GetDataList(ParamInfos.MarkLeaderLineEndPoint.Name, leaderLineEndPoints);
+
+            var textElements = new List<string>();
+            if (!dA.GetDataList(ParamInfos.Text.Name, textElements))
+                return null;
+
+            var attributes = dA.GetGooListValue<TSD.Text.TextAttributes>(ParamInfos.TextAttributes);
+
+            attributes = (attributes is null)
+                ?new List<TSD.Text.TextAttributes> { new TSD.Text.TextAttributes() } 
+                :attributes;
+
+            var textNumber = new int[]
+            {
+                insertionPoints.Count,
+                leaderLineEndPoints.Count,
+                textElements.Count,
+                attributes.Count
+            }.Max();
+
+            var insertedTexts = new TSD.Text[textNumber];
+            if (leaderLineEndPoints.Any())
+            {
+                CreateTextWithLeader(view,
+                    insertionPoints,
+                    leaderLineEndPoints,
+                    textElements,
+                    attributes,
+                    textNumber,
+                    insertedTexts);
+            }
+            else
+            {
+                CreateTextWithoutLeader(view,
+                    insertionPoints,
+                    textElements,
+                    attributes,
+                    textNumber,
+                    insertedTexts);
+            }
+            DrawingInteractor.CommitChanges();
+            dA.SetDataList(ParamInfos.Text.Name, insertedTexts);
+            return insertedTexts;
+        }
+
+        private void CreateTextWithoutLeader(TSD.View view, List<Point3d> insertionPoints, List<string> textElements, List<TSD.Text.TextAttributes> attributes, int textNumber, TSD.Text[] insertedTexts)
```

Comment:
Making return value _TSD.Text[] insertedTexts_ as one of the arguments makes the flow of this method too much complicated.
Please clearly specify method return type ( TSD.Text[] ). It will make your (this method) intent more obvious than it is now.